feat(minecraft-java) add wider packwiz support

H-QueerCoded opened the pull request:
**Description**

Updated minecraft-java's questions.yaml to allow downloading packwiz packs when modloaders other than Fabric are chosen. The developer of the docker image has [stated](https://github.com/itzg/docker-minecraft-server/issues/1238#issuecomment-1003560897) that packwiz support is not limited to Fabric. Mod loaders supported by packwiz itself are listed [here](https://packwiz.infra.link/reference/commands/packwiz/init/#options).

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

I don't have the capability to test this at this time.

**📃 Notes:**

I haven't worked on TrueNAS apps/questions.yaml before but based on my understanding from reading the docs this should be correct. If there is a way to combine all the added lines using a compound show-if statement (ie show-if  type is forge or fabric or quilt or neoforge) I couldn't find documentation on it.

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

H-QueerCoded committed:
Packwiz url for all supported loaders

and also custom

Signed-off-by: H <37481947+H-QueerCoded@users.noreply.github.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/truecharts/charts?pullRequest=17523) <br/>All committers have signed the CLA.

xstar97 commented:
Not how this works,  you can't dupe a variable like this

H-QueerCoded committed:
Rename minecraft-java Packwiz variables

Signed-off-by: H <37481947+H-QueerCoded@users.noreply.github.com>

H-QueerCoded commented:
Unsure if this change qualifies as minor or patch so I haven't changed the version yet

xstar97 commented:
```FABRIC_PACKWIZ_URL```
Is not a legit variable afaik from upstream docs

H-QueerCoded commented:
Oh yeah, you're right. The only variable is PACKWIZ_URL. Is there a way to do show-if such that one variable shows under various conditions? So there would be one PACKWIZ_URL variable that shows if a supported modloader is selected.

H-QueerCoded committed:
Revert "Rename minecraft-java Packwiz variables"

This reverts commit bd761c557ae1ad345b92f50e69c6bf66bd93ecba.

H-QueerCoded committed:
Revert "Packwiz url for all supported loaders"

This reverts commit 87f0575ebbadf7fd9421d8ee6db9094654a5bef2.

H-QueerCoded committed:
Rewrite PACKWIZ_URL changes

The option will now show for anything other than vanilla. Theoretically packwiz could be used for things beyond its supported mod loaders.

H-QueerCoded committed:
Update appVersion

Decided this change qualifies as a patch since it just changes an existing option

Signed-off-by: H <37481947+H-QueerCoded@users.noreply.github.com>

H-QueerCoded committed:
Revert "Update appVersion"

This reverts commit 510f875772bf53f0a25d5d9f3393bea097345c16.

H-QueerCoded committed:
Update minecraft-java version

changed the wrong value earlier

Signed-off-by: H <37481947+H-QueerCoded@users.noreply.github.com>

H-QueerCoded committed:
Merge branch 'master' into minecraft-java_questions_packwiz

H-QueerCoded committed:
Merge branch 'master' into minecraft-java_questions_packwiz

stavros-k reviewed (changes requested):
Options should be also present in the values.yaml.
Also PR title must follow formatting of all the other PRs
(Used by the changelog gneration)

xstar97 reviewed (changes requested):
@H-QueerCoded can you add the PACKWIZ_URL to the values.yaml and should be good to go.
default value should be blank; ""

H-QueerCoded committed:
Move PACKWIZ_URL up in values.yaml

Matches location in questions.yaml

H-QueerCoded commented:
It was already present but I relocated it farther up to match the position in questions.yaml

xstar97 commented:
bump the chart version ;)

H-QueerCoded committed:
Bump version

H-QueerCoded committed:
Merge branch 'master' into minecraft-java_questions_packwiz

Signed-off-by: H <37481947+H-QueerCoded@users.noreply.github.com>
